ui_framework/src/components/code_editor/code_editor.test.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import sinon from 'sinon';
+import { mount } from 'enzyme';
+import { KuiCodeEditor } from './code_editor';
+import { keyCodes } from '../../services';
+
+describe('KuiCodeEditor', () => {
+
+  let element;
+
+  beforeEach(() => {
```

Comment:
Can we put these tests inside another `describe` block, and add another test above it:

```
// You'll need to import this above.
import { requiredProps } from '../../test/required_props';

test('is rendered', () => {
  const component = <KuiCodeEditor />;
  expect(render(component)).toMatchSnapshot();
});
```

You'll need to run the tests again and commit the generated snapshot.

Reply:
Can you also add additional tests like this for `width` and `height`?

Reply:
I wasn't sure if we should do snapshot testing here, since that would also test, that `react-ace` doesn't change what they render. I am not sure, whether it would be good to add snapshots test upon the code generated by a library.

Reply:
Also for the `required_props` we would first need to look how we handle the props as discussed in the above comment.

Reply:
> I am not sure, whether it would be good to add snapshots test upon the code generated by a library.

I think it's OK to test library code if a component depends upon it. The test failure would be valuable because it will alert us to a change in DOM structure which would have otherwise gone unnoticed. This way, if some functional or other structure-reliant tests fail due to this change, we'll have a head's up and be able to clearly see how the structure changed. If the test failure is just noise and the change is superficial, then it's trivial to update the snapshot.